packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -2427,8 +2432,36 @@ function startTransition(
   }
 
   try {
-    setPending(false);
-    callback();
+    if (enableAsyncActions) {
+      const returnValue = callback();
+
+      // `isPending` is either `false` or a thenable that resolves to `false`,
+      // depending on whether the action scope is an async function. In the
+      // async case, the resulting render will suspend until the async action
+      // scope has finished.
+      const isPending = requestAsyncActionContext(returnValue);
+      setPending(isPending);
+    } else {
+      // Async actions are not enabled.
+      callback();
+      setPending(false);
```

Comment:
Does the order of executing the callback and setting the flag matter? Mostly asking because this order is changed now even with the feature flag turned off.

Reply:
Yeah that's a good point, I suppose it should come before so that the flag gets turned off even if there's an error. Looks like we don't have a test for that.